frame/im-online/src/lib.rs

```diff
@@ -391,13 +391,13 @@ pub mod pallet {
 			heartbeat.validators_len as u32,
 			heartbeat.network_state.external_addresses.len() as u32,
 		))]
-		pub fn heartbeat(
+		pub(crate) fn heartbeat(
```

Comment:
Why? Why should they be only crate local? Please revert these changes.

Reply:
I don't have any preference of `pub(crate)` or `pub`. It's just that before the pallet macro migration these functions are only crate local, and I changed them `pub` in the migration PRs, like this one https://github.com/paritytech/substrate/pull/8714/files#diff-9fb584356be57c2292f7f3ba4197a2a65ba1ac27c2c4b076d3584cc625dd9d83L354-L358 And I was just trying to make the visibility as originally crate local, no matter whatever reason they were defined as this.

If you prefer to let them be `pub` instead, I'm happy to revert these changes.

Reply:
In the end they are pub anyway through the `Call` enum as mentioned by @shawntabrizi. And maybe some people rely on them being public now. In the end there is no downside of having them public

Reply:
Reverted.